builtin_object.go

```diff
@@ -283,23 +283,16 @@ func (r *Runtime) object_freeze(call FunctionCall) Value {
 		obj.self.preventExtensions(true)
 
 		for item, next := obj.self.iterateKeys()(); next != nil; item, next = next() {
-			if prop, ok := item.value.(*valueProperty); ok {
-				prop.configurable = false
-				if !prop.accessor {
-					prop.writable = false
-				}
+			prop := obj.getOwnProp(item.name)
+			descr := PropertyDescriptor{
+				Configurable: FLAG_FALSE,
+			}
+			if prop, ok := prop.(*valueProperty); ok && prop.accessor {
+				// no-op
 			} else {
-				prop := obj.getOwnProp(item.name)
-				descr := PropertyDescriptor{
-					Configurable: FLAG_FALSE,
-				}
-				if prop, ok := prop.(*valueProperty); ok && prop.accessor {
-					// no-op
-				} else {
-					descr.Writable = FLAG_FALSE
-				}
-				obj.defineOwnProperty(item.name, descr, true)
+				descr.Writable = FLAG_FALSE
 			}
+			obj.defineOwnProperty(item.name, descr, true)
```

Comment:
This is just what the specification says - I think the previous code was making it faster in some cases but I couldn't figure out how to fix it in any other way 🤷‍♂and had other problems ;)

This is problematic for the way I have implemented modules namespace, which might be the actual problem 🤷‍♂ but this was in reality one of the places where it had problems and both were fixed by doing exactly what the specification says.